### ISSUE TITLE
Request: Sequential download instead of parrallel?

### ISSUE THREAD

**Issue by grimson73** [zeld/psa-update#10]
```
Check for a map update (hit ESC to skip): Europe
Update available - Type: Map       Version: 16.0.0-r0
                   Size: 19.82 GB  Release date: 2024-03-13 19:32:31.0
                   URL: https://download.tomtom.com/OEM/PSA/MAP/PSA_map-eur_16.0.0-r0-NAC_EUR_WAVE4.tar
Download update? yes
Update available - Type: Firmware  Version: 44.07.05.22_NAC-r0
                   Size: 6.31 GB   Release date: 2023-07-03 10:59:21.0
                   URL: https://majestic-web.mpsa.com/mjf00-web/rest/UpdateDownload?updateId=0000000000000000000&uin=000000000000000000000&type=fw
                   License URL: https://majestic-web.mpsa.com/mjf00-web/rest/LicenseDownload?mediaVersion=000000000000000000&uin=000000000000000000000
Download update? yes
 30% [######>-------------] 9.94 MiB/s   ETA=22m PSA_map-eur_16.0.0-r0-NAC_EUR_WAVE4.tar
100% [####################] 0 B/s        ETA=0s  licenses_0000000000000000_20240924161735.tar
 10% [#>------------------] 617.17 KiB/s ETA=3h  PSA_ovip-int-firmware-version_44-07-05-22_NAC-r0_NAC_EUR_WAVE4.tar        
```
Hi thanks for this excellent alternative update/download app.
At first my downloads timed out because of a busy disk (extracting other files) so the above got me thinking. I personally prefer to download files in sequence, so all bandwidth is used only per file. As the example shows the firmware file is lagging because of multiple downloads. As I type this, I notice I could choose to download per file (not choosing yes on all files at once) but I do want to make a request :)

Would it be possible to download sequentially when choosing all files at once? Again, I know I can choose per file yes, but it might happen you want to download them all in one batch and not check on when to download the next etc.
Thanks!!

**Comment by zeld** [comment]
Hi,

The original reason for the parallel download is because stellantis server (from where firmware is downloaded) is painfully slow, contrarily to tomtom server.

I don't promise anything but I can have a look sometime to add a flag to download files sequentially, it should not be complex to do. So leaving the request open for now.

As a workaround you could also use the non interactive mode and sequentially download files: ./psa-update --silent VIN && ./psa-update --silent VIN --map eur

**Comment by grimson73** [comment]
> The original reason for the parallel download is because stellantis server (from where firmware is downloaded) is painfully slow, contrarily to tomtom server.

> As a workaround you could also use the non interactive mode and sequentially download files: ./psa-update --silent VIN && ./psa-update --silent VIN --map eur

Thanks for the explanation and the 'workaround'. In hindsight it's really a non-issue to be honest :) .. but thanks for listening for feedback. Please close if not worth the time developing this request :).

**Comment by zeld** [comment]
Introduced argument `--sequential-download` to force sequential download of updates. This is available in release 1.0.6.
